.NET/Microsoft.Recognizers.Text.NumberWithUnit/Extractors/BaseMergedUnitExtractor.cs

```diff
@@ -216,5 +220,30 @@ private void MergePureNumber(string source, List<ExtractResult> ers)
 
             ers.Sort((x, y) => x.Start - y.Start ?? 0);
         }
+
+        // Add multiplier to extraction when it follows the unit e.g. "10 USD million"
+        private void MergeMultiplier(string source, List<ExtractResult> ers)
```

Comment:
Wouldn't it be best for perf to treat such cases as "special units", but with special resolution to convert back to the base currency? It can even sometimes show up as, for example, "MUSD" - as in "Direct support to the armed forces, totalling 50 MUSD, has been allocated in the budget".

Also, as far as I'm aware, such mention use is mostly common in finance, so it should probably be blocked for other units.

Reply:
mmh, if I understand correctly, the main difference would be to replace the regex search with the TrieTree search.
But since the multiplier regex is small, I do not think the perf impact is noticeable. 
I will try to add the multipliers as special units to the CurrencySuffixList and check.

Yes, that is already the case. MergeMultiplier is inside MergeCompoundUnits which is called only for Currency units.

Reply:
Treating them as special units (where each suffix unit is combined with each multiplier in a new suffix dictionary) increases execution time by 28%.
Using the current approach with regex, the impact is 0.87% (comparable to fluctuations between different executions: 0.84%).
One could also treat only the multipliers as special units but then the processing would be the same as the one already implemented and there would be little difference in performance.

Reply:
Thanks for checking this. Seems good to me then. We'll continue the review soon.